IDE/QNX/example-client/client-tls.c

```diff
@@ -0,0 +1,261 @@
+        free(pem);
+        return -1;
+    }
+    *out = pem;
+    return pemSz;
+}
+
+
+int main(int argc, char** argv)
+{
+    int                sockfd;
+    struct sockaddr_in servAddr;
+    char               buff[256];
+    size_t             len;
+    int                ret;
+    int                port;
+
+    int pemSz;
+    unsigned char *pem;
+    FILE* f;
+
+    /* declare wolfSSL objects */
+    WOLFSSL_CTX* ctx;
+    WOLFSSL*     ssl;
+
+    /* Check for proper calling convention */
+    if (argc != 6) {
+        printf("usage: %s <port> <IPv4 address> <CA PEM> <cert DER> <key DER>\n",
+                argv[0]);
+        return 0;
+    }
+    port = atoi(argv[1]);
+
+    /* Create a socket that uses an internet IPv4 address,
+     * Sets the socket to be stream based (TCP),
+     * 0 means choose the default protocol. */
+    if ((sockfd = socket(AF_INET, SOCK_STREAM, 0)) == -1) {
+        fprintf(stderr, "ERROR: failed to create the socket\n");
+        ret = -1;
+        goto end;
+    }
+
+    /* Initialize the server address struct with zeros */
+    memset(&servAddr, 0, sizeof(servAddr));
+
+    /* Fill in the server address */
+    servAddr.sin_family = AF_INET;             /* using IPv4      */
+    servAddr.sin_port   = htons(port); /* on DEFAULT_PORT */
+
+    /* Get the server IPv4 address from the command line call */
+    if (inet_pton(AF_INET, argv[2], &servAddr.sin_addr) != 1) {
+        fprintf(stderr, "ERROR: invalid address\n");
+        ret = -1;
+        goto end;
+    }
+
+    /* Connect to the server */
+    if ((ret = connect(sockfd, (struct sockaddr*) &servAddr, sizeof(servAddr)))
+         == -1) {
+        fprintf(stderr, "ERROR: failed to connect\n");
+        goto end;
+    }
+
+    /*---------------------------------*/
+    /* Start of security */
+    /*---------------------------------*/
+    /* Initialize wolfSSL */
+    if ((ret = wolfSSL_Init()) != WOLFSSL_SUCCESS) {
+        fprintf(stderr, "ERROR: Failed to initialize the library\n");
+        goto socket_cleanup;
+    }
+
+    /* Create and initialize WOLFSSL_CTX */
+    if ((ctx = wolfSSL_CTX_new(wolfTLSv1_2_client_method())) == NULL) {
+        fprintf(stderr, "ERROR: failed to create WOLFSSL_CTX\n");
+        ret = -1;
+        goto socket_cleanup;
+    }
+
+    /* load cert and convert DER to PEM using dynamic length */
+    pemSz = convertDerToPem(CERT_TYPE, argv[4], &pem);
+    if (pemSz <= 0) {
+        fprintf(stderr, "ERROR: converting DER cert to PEM\n");
+        ret = -1;
+        goto socket_cleanup;
+    }
+
+    if (wolfSSL_CTX_use_certificate_buffer(ctx, pem, pemSz,
+                WOLFSSL_FILETYPE_PEM) != WOLFSSL_SUCCESS) {
+        fprintf(stderr, "issue loading in pem cert\n");
+        ret = -1;
+        free(pem);
+        goto socket_cleanup;
+    }
+    free(pem);
+
+    /* load key and convert DER to PEM using dynamic length */
+    pemSz = convertDerToPem(PRIVATEKEY_TYPE, argv[5], &pem);
```

Comment:
I see your arguments expect DER, but not sure why convert to PEM and back to DER again? Are you able to use the DER directly and pass `WOLFSSL_FILETYPE_ASN1` below for `wolfSSL_CTX_use_PrivateKey_buffer`?

Reply:
The extra conversions from DER/PEM was done for a sanity check and example of the conversion. During the development it had come up.